dbms/src/Interpreters/Join.cpp

```diff
@@ -1123,17 +1199,32 @@ void NO_INLINE joinBlockImplTypeCase(
         {
             auto key_holder = key_getter.getKeyHolder(i, &pool, sort_key_containers);
             auto key = keyHolderGetKey(key_holder);
-            size_t segment_index = 0;
             size_t hash_value = 0;
-            if (map.getSegmentSize() > 0 && !ZeroTraits::check(key))
+            size_t segment_index = 0;
+            if (enable_fine_grained_shuffle)
```

Comment:
Can we still calculate the hash value for fine grained shuffle? 
Because that would mislead the hash value for fine grained shuffle is not necessary.